### PR TITLE
Add a 'dome' attribute to Observatory.

### PR DESCRIPTION
**Pull request by jamessynge**
Add logger kwarg to create_dome_from_config, enabling caller
to provide logger to be used.

### CODE REVIEW

File: pocs/observatory.py [hunk 1]
```diff
@@ -488,6 +498,8 @@ def _create_mount(self, mount_info=None):
         Note:
             This does not actually make a serial connection to the mount. To do so,
             call the 'mount.connect()' explicitly.
+            TODO(jamessynge): Discuss this claim with Wilfred. SerialData automatically
```
Comment (wtgee) [inline pocs/observatory.py]: The claim is wrong as you point out and should be updated.

File: pocs/dome/__init__.py [hunk 2]
```diff
@@ -30,10 +26,10 @@ def create_dome_from_config(config):
     else:
         brand = dome_config.get('brand')
         driver = dome_config['driver']
```
Comment (wtgee) [inline pocs/dome/__init__.py]: This already made it's way in, but this will fail if `dome` is in config but `driver` is not.

Comment (wtgee) [reply pocs/dome/__init__.py]: Perhaps that was intended but if so we should catch and warn.

Comment (jamessynge) [reply pocs/dome/__init__.py]: Good point. This is certainly a fatal error in terms of creating the dome instance, and will most likely keep POCS from working as intended. Probably we should just raise an exception.

File: pocs/observatory.py [hunk 3]
```diff
@@ -128,6 +135,9 @@ def status(self):
                     status['mount']['mount_target_ha'] = self.observer.target_hour_angle(
                         t, self.mount.get_target_coordinates())
 
+            if self.dome:
+                status['dome'] = self.dome.status
```
Comment (wtgee) [inline pocs/observatory.py]: I realize this has already been merged, but it's a little odd that `status` is a property here but a method for everything else. Why not make the dome status a method?

**Pull request merged**